Feature user

luz315 opened the pull request:
## PR 체크리스트
-  커밋 컨벤션에 맞게 작성했는가?
-  PR 전에 현재 브랜치를 pull 받았는가?
-  구현한 기술들을 문서화 하였는가?


<br>

## PR 작업 분류

<!-- Please check the one that applies to this PR using "x". -->

- [x] 신규 기능
- [ ] 버그 수정
- [x] 리팩토링
- [ ] 기타


<br>

## 작업 상세 내용
- 카프카 단방향 통신(유저 변동 감지 시 auth로 이벤트)
- 유저 생성, 삭제 마스터 관리자 권한 분리
- 유저 스트림 완성
- 유저 사용자 정보 파싱, 권한
- 유저 마스터 권한 커스텀 AOP 로직
  - 헤더에서 이메일, 롤 값 파싱
  - 헤더 인터셉터 구현
  - Baseentity Auditing적용
  - 스레드 로컬을 통해 유저 인자 없이 자동 파악 시스템
- 유저 에러핸들링 구현
- 유저 검색로직 구현
  - 검색 전용 디티오 구현
  - queryDSL 적용
  - 커스텀 페이지네이션 모델 구현
- 유저 엔티티, 디티오 구현
- 유저 권한 승인 처리 로직
  - 기존: 유저 레디스 스트림 유저 생성 시에 전송
  - 변경: 유저 레디스 스트림 마스터가 유저 생성 시 & 유저 승인 시에 전
- 유저 컨트롤러 마스터 & 일반 유저 분리
- 유저 서비스 DIP 준수 로직 분리

<br>

## 🔴 이건 반드시 확인해 주세요!

luz315 committed:
[feat] git init

luz315 committed:
[feat] user 도메인 완성

luz315 committed:
[feat] user 어플리케이션 계층 완성

- 디티오, 서비스, 인터페이스 구현체

luz315 committed:
[feat] user 인프라 계층 완성

- 카프카, 시큐리티, 리포지토리

luz315 committed:
[feat] user 프레젠테이션 계층 완성

asqwklop12 commented on user/src/main/java/com/sparta/rooibos/user/application/dto/UserRequest.java:
```diff
@@ -0,0 +1,34 @@
+        String email,
+
+        @NotBlank(message = "비밀번호는 필수입니다.")
+        @Size(min = 4, message = "비밀번호는 최소 4자 이상이어야 합니다.")
```
메시지가 요구사항에 맞지 않습니다.
요구사항에는 최소 8자 이상, 15자 이하이며 알파벳 대소문자(a~z, A~Z), 숫자(0~9), 특수문자으로 명시되어있습니다.

asqwklop12 replied:
비밀번호 요구사항입니다.

asqwklop12 commented on user/src/main/java/com/sparta/rooibos/user/application/service/UserService.java:
```diff
@@ -0,0 +1,94 @@
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("해당 사용자를 찾을 수 없습니다."));
+
+        return ResponseEntity.ok("유저가 조회되었습니다.");
```
이거 리턴 타입이 이상한데 확인부탁드리겠습니다.

asqwklop12 commented on user/src/main/java/com/sparta/rooibos/user/infrastructure/security/SecurityConfig.java:
```diff
@@ -0,0 +1,52 @@
+        http
+                .cors(corsCustomizer -> corsCustomizer.configurationSource(request -> {
+                    CorsConfiguration configuration = new CorsConfiguration();
+                    configuration.setAllowedOrigins(Collections.singletonList("http://localhost:3000"));
```
이거 머예여 저희 3000번 포트 안쓰는데...

asqwklop12 commented on user/src/main/java/com/sparta/rooibos/user/domain/entity/User.java:
```diff
@@ -0,0 +1,56 @@
+
+@Entity
+@Getter
+@Table(name = "p_users")
```
저희 단수인걸로 기억하는데... ㅜㅜ
https://www.notion.so/teamsparta/ERD-1b32dc3ef5148025b772dd7b14d16d5c?pvs=4#1b32dc3ef514800782d1ef9b3d6b7697

asqwklop12 reviewed (commented):
수고하셨습니다.!!

luz315 committed:
Merge branch 'develop' of https://github.com/Sparta-Rooibos/rooibos into feature_user_test

luz315 committed:
[refactor] 유저 스트림 완성

luz315 committed:
[refactor] 유저 사용자 정보 파싱, 권한

- 유저 마스터 권한 커스텀 AOP 로직
- 헤더에서 이메일, 롤 값 파싱
- 헤더 인터셉터 구현
- Baseentity Auditing적용
- 스레드 로컬을 통해 유저 인자 없이 자동 파악 시스템

luz315 committed:
[refactor] 유저 에러핸들링 구현

luz315 committed:
[refactor] 유저 검색로직 구현

- 검색 전용 디티오 구현
- queryDSL 적용
- 커스텀 페이지네이션 모델 구현

luz315 committed:
[refactor] 유저 에러핸들링 구현

luz315 committed:
[refactor] 유저 엔티티, 디티오 구현

luz315 committed:
[refactor] 유저 권한 승인 처리 로직

- 기존: 유저 레디스 스트림 유저 생성 시에 전송
- 변경: 유저 레디스 스트림 마스터가 유저 생성 시 & 유저 승인 시에 전

luz315 committed:
[refactor] 유저 컨트롤러 마스터 & 일반 유저 분리

luz315 committed:
[refactor] 유저 서비스 DIP 준수 로직 분리
